source/auth/tests/mongodb-oidc.rst

```diff
@@ -97,53 +112,110 @@ verify usage.  Unless otherwise specified, the tests MUST be performed with
 the authorization code workflow with and without a provided refresh callback.
 If desired, the caching tests can be done using mock server responses.
 
```

Comment:
1. Done
2. I moved that sentence to the top level notes
3. I removed the note about overriding creds
4. That sentence was a mistake, the tests should provide both callbacks unless otherwise noted.  Updated
5. I added a clarifying sentence
6. I switched the terminology to use "callback-driven"
7. Done